pkg/v1/task/task.go

```diff
@@ -159,6 +154,19 @@ func makeContractTaskOpts(t *TaskShared, taskDefaults *create.TaskDefaults) *con
 		taskOpts.ScheduleTimeout = &scheduleTimeout
 	}
 
+	// Only set Retries if it's not nil
+	if t.Retries != nil {
+		taskOpts.Retries = *t.Retries
+	}
+
+	if t.RetryBackoffFactor != nil {
```

Comment:
Discovered that `RetryBackoffFactor` and `RetryMaxBackoffSeconds` weren't getting passed through on the Go SDK -- moved the `t.Retries` logic down here to keep it consolidated